macros/utils/table_operations/replace_table_data.sql

```diff
@@ -38,3 +38,10 @@
 
     {% do adapter.drop_relation(intermediate_relation) %}
 {% endmacro %}
+
+{% macro athena__replace_table_data(relation, rows) %}
+     {% call statement('truncate_relation') -%}
+        delete from {{ relation }}
```

Comment:
I wish:
```
truncate table schema_columns_snapshot;
```
Athena:
```
Queries of this type are not supported
```
This could be improved if temporary tables are switched from hive to iceberg to use table renaming possible for iceberg tables (the second known issue in the pr description)

Reply:
`delete from my_table;`  when using iceberg is the same as a truncate in pg, delete will just mark objects with the delete markers, but it doesn't delete any data. Data will be deleted when calling procedures like vacuum (that calls expire snapshots)